### ISSUE TITLE
[bug] PHP 8.4 Deprecated Error

### ISSUE THREAD

**Issue by moneysavingpro** [serbanghita/Mobile-Detect#972]
Since upgrading to PHP 8.4 I'm now receiving a deprecated error:

> PHP Deprecated: Detection\MobileDetect::__construct(): Implicitly marking parameter $cache as nullable is deprecated, the explicit nullable type must be used instead in /vendor/mobiledetect/mobiledetectlib/src/MobileDetect.php on line...

**Comment by BackuPs** [comment]
same issue here... i had the wrong version loaded.  I do not get the warning deprecated message anymore .  Sorry

**Comment by serbanghita** [comment]
```
PHP 8.4.1 (cli) (built: Nov 25 2024 18:03:47) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.4.1, Copyright (c) Zend Technologies
    with Zend OPcache v8.4.1, Copyright (c), by Zend Technologies
    with Xdebug v3.4.0, Copyright (c) 2002-2024, by Derick Rethans
```

```
error_reporting = E_ALL
display_errors = On
````

and I can't reproduce with the latest version https://github.com/serbanghita/Mobile-Detect/releases/tag/4.8.09

@moneysavingpro what version are you using exactly? BTW in production is not okay have the `DEPRECATED` flag enabled

**Comment by moneysavingpro** [comment]
@serbanghita this has been fixed since upgrading to 4.8.0. thank you!
